src/plugins/intel_cpu/src/nodes/executors/dnnl/dnnl_matmul_primitive.cpp

```diff
@@ -415,6 +422,8 @@ static std::pair<VectorDims, VectorDims> makeDummyInputDims(const Shape& in0,
                 } else {
                     inDims1[idx1] = inDims0[idx0];
                 }
+            } else if (inDims0[idx0] != Shape::UNDEFINED_DIM && inDims1[idx1] != Shape::UNDEFINED_DIM) {
```

Comment:
This fix looks strange.
How is this case even possible?
My guess is that it is because we do not consider `weightsNonTransposed` here (see my other comment)